test/wpackettest.c

```diff
@@ -349,6 +353,43 @@ static int test_WPACKET_memcpy(void)
     return 1;
 }
 
+static int test_WPACKET_init_der(void)
+{
+    WPACKET pkt;
+    unsigned char sbuf[1024];
+    unsigned char testdata[] = { 0x00, 0x01, 0x02, 0x03 };
+    unsigned char testdata2[259]  = { 0x82, 0x01, 0x00 };
+    size_t written;
+
+    /* Test initialising for writing DER */
+    if (!TEST_true(WPACKET_init_der(&pkt, sbuf, sizeof(sbuf)))
+            || !TEST_true(WPACKET_put_bytes_u24(&pkt, 0xfffefd))
+               /* Test writing data in a length prefixed sub-packet */
+            || !TEST_true(WPACKET_start_sub_packet(&pkt))
+            || !TEST_true(WPACKET_memcpy(&pkt, testdata, sizeof(testdata)))
+            || !TEST_true(WPACKET_close(&pkt))
+            || !TEST_true(WPACKET_put_bytes_u8(&pkt, 0xfc))
+            || !TEST_true(WPACKET_finish(&pkt))
+            || !TEST_true(WPACKET_get_total_written(&pkt, &written))
+            || !TEST_mem_eq(WPACKET_get_curr(&pkt), written, simpleder,
+                            sizeof(simpleder)))
+        return cleanup(&pkt);
```

Comment:
Would you mind writing in a comment what ASN.1 structure you're writing here?  I do not understand what you're doing here from a structural point of view, or how to draw actual examples from this.